### PR TITLE
Add supervisor

### PR DESCRIPTION
**Pull request by Velua**

### CODE REVIEW

File: services/user/App1/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import "./App.css";
+import { connect } from "@psibase/plugin";
```
Comment (James-Mart) [inline services/user/App1/src/App.tsx:2]: This should come from `@psibase/common-lib` or `/common`

File: libraries/psibase/tester/src/DefaultTestChain.cpp
Comment (James-Mart) [inline libraries/psibase/tester/src/DefaultTestChain.cpp:1]: This is the file that is used for setting up the chain that the tester uses. The tester is what runs the C++ service unit tests. I don't think we need to update the tester chain with any of our POC code.

File: services/system/AccountSys/ui/src/App.tsx [hunk 2]
```diff
@@ -39,6 +43,23 @@ export interface AccountWithKey extends AccountWithAuth {
     privateKey: KeyPairWithAccounts["privateKey"];
 }
 
+/// Problem: All these exisitng system / user apps are using the old operations / mutations or whatever
+// Solution: Migrate them all to Plugin WASMs, and figure out
+// Importables! -> Providing a JS function to the WASM for it to call, to add actions to the actions array,
+// We don't solely operate on the return values of these WASM functions, we also require to monitor the functions that are called (that we give it via importables)
```
Comment (James-Mart) [inline services/system/AccountSys/ui/src/App.tsx:49]: Idk what you mean we are required to monitor the functions that are called

File: services/system/AccountSys/ui/src/App.tsx [hunk 3]
```diff
@@ -39,6 +43,23 @@ export interface AccountWithKey extends AccountWithAuth {
...
+//
+// Do we have operations that do network requests prior to providing actions to submit to chain?
+// JS Operations -> Can do network requests
+// WASM operations / functions -> Can't.
+//
+// Solution 1: Provide Rust with the option of network calls via importables
+// Solution 2: Figger it out in the app rather than operations... and dumb it down for the WASM
+//
+// In the future, we want these WASM functions to know context, e.g. who called this? and what app they were in when it was called?
```
Comment (James-Mart) [inline services/system/AccountSys/ui/src/App.tsx:58]: This will be done via an importable like, "getSender". The loader knows who called it because that information was sent by the supervisor, so it can fill in the getSender importable with the name of the caller service.

File: services/user/App1/src/App.tsx (hunk 1 above)
Comment (Velua) [reply services/user/App1/src/App.tsx:2]: Why can't we use the external package if other app developers are gonna use it?

Comment (James-Mart) [reply services/user/App1/src/App.tsx:2]: I just meant that this API is part of the common libraries. And since supervisor does more than just plugin communication, I think it should be called "common-lib" or "common" rather than "plugin"

File: services/user/App1/src/App.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+function App() {
+  const run = async () => {
+    const res = await connect();
+
+    console.log(res, "came back");
+    const back = await res.functionCall({
+      service: "app2",
```
Comment (cool-ant) [inline services/user/App1/src/App.tsx:10]: `plugin` instead of `service`? so that `method` doesn't sound like `action`?

Comment (cool-ant) [reply services/user/App1/src/App.tsx:10]: `account`?
`plugin-method`?

File: services/user/TokenSys/wasm/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+impl Guest for Component {
+    /// Say hello!
+    fn transfer(to: String, amount: u32) -> String {
+        let logged_in_user = imports::get_logged_in_user();
+
+        println!("logged_in_user x, 'is a thing.. {:?}", logged_in_user);
+
+        let test_params = Funccallparams {
+            service: "test_service".to_string(),
+            method: "test_method".to_string(),
```
Comment (cool-ant) [inline services/user/TokenSys/wasm/src/lib.rs:20]: should this be `action`, not `method`?

File: services/user/App1/src/App.tsx (hunk 4 above)
Comment (cool-ant) [reply services/user/App1/src/App.tsx:10]: and perhaps `functionCall` --> `pluginFnCall`?
